### PR TITLE
Add support for limit_req_zone in main nginx config and limit_req: Fixes #1134

### LINKED ISSUE #1134: Feature request: rate limiting (limit_req_zone, limit_req)
(D)DoS, brute force attacks etc are everywhere.
We want to protect our backends from excessive requests and load.
Nginx is supporting rate limiting in the `ngx_http_limit_req_module` module:
https://www.nginx.com/blog/rate-limiting-nginx/

Bring this feature as default to the puppet nginx core and help people to protect their backends.

### PR DESCRIPTION
**Pull request by absltkaos**
Add support for `limit_req_zone` in main nginx config and `limit_req`
for `nginx::resource::location`.

In init.pp
`limit_req_zone` can be a String, or an array of String

In resource/location.pp
`limit_zone` can be a String and should point to a zone defined from
`limit_req_zone` in init.pp

### CODE REVIEW

File: manifests/init.pp [hunk 1]
```diff
@@ -45,6 +45,7 @@
   $global_owner                                              = $nginx::params::global_owner,
   $global_group                                              = $nginx::params::global_group,
   $global_mode                                               = $nginx::params::global_mode,
+  Optional[Variant[String, Array[String]]] $limit_req_zone   = undef,
```
Comment (bastelfreak) [inline manifests/init.pp]: An empty string isn't valid here, right? So we could prohibit it:
```suggestion
  Optional[Variant[String[1], Array[String[1]]]] $limit_req_zone = undef,
```

File: manifests/resource/location.pp [hunk 2]
```diff
@@ -214,6 +216,7 @@
   Boolean $ssl                                                     = false,
   Boolean $ssl_only                                                = false,
   Optional[String] $location_alias                                 = undef,
+  Optional[String] $limit_zone                                     = undef,
```
Comment (bastelfreak) [inline manifests/resource/location.pp]: Same here, I guess empty strings aren't allowed:
```suggestion
  Optional[String[1]] $limit_zone                                  = undef,
```

**Comment by bastelfreak** [comment]
Hi @absltkaos, thanks for the PR. can you add some unit tests for it?

**Comment by absltkaos** [comment]
> Hi @absltkaos, thanks for the PR. can you add some unit tests for it?

I'm not super familiar with Rspec, but I'll see what I can do.

**Comment by bastelfreak** [comment]
thanks!

**Pull request merged**